### ISSUE TITLE
Unexpected result for SDK replace method with watch=True

### ISSUE THREAD

**Issue by bartgras** [kserve/kserve#1739]
/kind bug

**What steps did you take and what happened:**
Used Python SDK to:
* Execute SDK `create` method to create isvc and wait until it's available
* Execute SDK `replace` method (with `watch=True`) to create new revision

**What did you expect to happen:**
In SDK `replace` method with `watch=True` I'd expect to wait for the new revision to become available. Instead watch immediately returns with `READY` equal to `True` because isvc was already available from the previous revision.

The wording in SDK documentation for any non-`create` method that accepts `watch=True` is a bit misleading. What would be the case for using `watch=True` in `replace`, `patch`, `promote` etc.?

**Anything else you would like to add:**
I'd like to understand how in general one can (with SDK) `create` isvc and then `replace` it while checking if and when new revision becomes available. 

In other words, how can one use SDK to create pinned or canary rollout and observe new revision's progress.

**Comment by bartgras** [comment]
I've noticed that after calling `replace` isvc is going to `READY` == `False` while previous revision is still available for predictions.

Quick solution to the issue would be to wait for a few seconds and then call [isvc_watch](https://github.com/kubeflow/kfserving/blob/1381f86e9b6ad831416008f5d25f0ed693f70871/python/kfserving/kfserving/api/kf_serving_client.py#L258). Seems a bit flaky though.
